pkg/eventcache/eventcache.go

```diff
@@ -131,21 +129,29 @@ func (ec *Cache) handleEvents() {
 		// process and lets copy it across.
 		if event.internal == nil {
 			event.internal, err = event.msg.RetryInternal(event.event, event.startTime)
+			if err != nil && event.color >= CacheStrikes {
+				if errors.Is(err, ErrFailedToGetParentInfo) {
+					errormetrics.ErrorTotalInc(errormetrics.EventCacheParentInfoFailed)
+				} else if errors.Is(err, ErrFailedToGetProcessInfo) {
+					errormetrics.ErrorTotalInc(errormetrics.EventCacheProcessInfoFailed)
+					eventcachemetrics.ProcessInfoError(notify.EventTypeString(event.event)).Inc()
+				}
+			}
 		}
 		if err == nil {
 			err = event.msg.Retry(event.internal, event.event)
+			if err != nil && event.color >= CacheStrikes {
+				if errors.Is(err, ErrFailedToGetPodInfo) {
+					eventcachemetrics.PodInfoError(notify.EventTypeString(event.event)).Inc()
+				}
+			}
 		}
 		if err != nil {
 			event.color++
 			if event.color < CacheStrikes {
 				tmp = append(tmp, event)
 				continue
 			}
-			if errors.Is(err, ErrFailedToGetProcessInfo) {
-				eventcachemetrics.ProcessInfoError(notify.EventTypeString(event.event)).Inc()
-			} else if errors.Is(err, ErrFailedToGetPodInfo) {
-				eventcachemetrics.PodInfoError(notify.EventTypeString(event.event)).Inc()
-			}
```

Comment:
I thought since ```err``` variable passes through 2 functions ```event.msg.RetryInternal(``` && ```event.msg.Retry(``` meanwhile we must consider it at each call